dequeue/dequeue.go

```diff
@@ -97,8 +98,11 @@ func (d *dequeuer) Dequeue(ctx context.Context) error {
 		msgs, err := sub.Pop(d.batchSize, nats.Context(tctx))
```

Comment:
I'm wondering if we entirely want to remove the context timeout and use `ctx` here directly. Need to think about why we want to cancel this `Pop` operation. I guess it's because we want to catch slow connections where NATS is failing to tell us anything. If we didn't put a deadline on that operation, would we _not_ see any error at all? Need to investigate this.